src/m365/spo/commands/site/site-admin-list.ts

```diff
@@ -0,0 +1,262 @@
+
+      if (responseContent.ErrorInfo) {
+        throw responseContent.ErrorInfo.ErrorMessage;
+      }
+
+      const sites: SPOSitePropertiesEnumerable = jsonData[jsonData.length - 1];
+      const siteId = sites?._Child_Items_?.[0]?.SiteId ?? undefined;
+
+      if (!siteId) {
+        throw 'Failed to obtain site Id from the provided site URL.';
+      }
+
+      const guid = siteId.replace('/Guid(', '').replace(')/', '');
+
+      if (!validation.isValidGuid(guid)) {
+        throw 'Failed to obtain site Id from the provided site URL.';
+      }
+      return guid;
+    }
+    catch (error) {
+      throw error;
+    }
+  }
+
+  private async getPrimaryAdminLoginNameFromAdmin(adminUrl: string, siteId: string): Promise<string> {
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${adminUrl}/_api/SPO.Tenant/sites(%27${siteId}%27)?$select=OwnerLoginName`,
+        headers: {
+          accept: 'application/json;odata=nometadata',
+          'content-type': 'application/json;charset=utf-8'
+        }
+      };
+
+      const response: string = await request.get<string>(requestOptions);
+      const responseContent = JSON.parse(response);
+      return responseContent.OwnerLoginName;
+    }
+    catch (err: any) {
+      throw err;
+    }
+  }
+
+  private async callAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      await logger.logToStderr('Retrieving site administrators...');
+      const requestOptions: CliRequestOptions = {
+        url: `${args.options.siteUrl}/_api/web/siteusers?$filter=IsSiteAdmin%20eq%20true`,
+        method: 'GET',
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      const responseContent: SiteResult = await request.get<SiteResult>(requestOptions);
+      const primaryOwnerLogin = await this.getPrimaryOwnerLoginFromSite(args.options.siteUrl);
+      const mappedResult = responseContent.value.map((u: SiteUserResult): CommandResultItem => {
+        return {
+          Id: u.Id,
+          LoginName: u.LoginName,
+          Title: u.Title,
+          PrincipalType: u.PrincipalType,
+          PrincipalTypeString: ListPrincipalType[u.PrincipalType],
+          Email: u.Email,
+          IsPrimaryAdmin: u.LoginName === primaryOwnerLogin
+        };
+      });
+      await logger.log(mappedResult);
+    }
+    catch (err: any) {
+      throw err;
+    }
+  }
+
+  private async getPrimaryOwnerLoginFromSite(siteUrl: string): Promise<string> {
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${siteUrl}/_api/site/owner`,
+        method: 'GET',
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
+        responseType: 'json'
+      };
+
+      const responseContent = await request.get<{ LoginName: string }>(requestOptions);
+      return responseContent?.LoginName ?? undefined;
```

Comment:
I think it may be cleaner if we add the ` | null` as possible result and return `null` instead of `undefined` if we won't find the `LoginName`. What do you think?

```suggestion
  private async getPrimaryOwnerLoginFromSite(siteUrl: string): Promise<string | null> {
    try {
      const requestOptions: CliRequestOptions = {
        url: `${siteUrl}/_api/site/owner`,
        method: 'GET',
        headers: {
          'accept': 'application/json;odata=nometadata'
        },
        responseType: 'json'
      };

      const responseContent = await request.get<{ LoginName: string }>(requestOptions);
      return responseContent?.LoginName ?? null;
```

Reply:
Done